wsd/COOLWSD.cpp

```diff
@@ -3274,6 +3277,103 @@ class ClientRequestDispatcher final : public SimpleSocketHandler
         LOG_INF("Sent capabilities.json successfully.");
     }
 
+    void handleCheckRequest(const Poco::Net::HTTPRequest& request,
+                                   const std::shared_ptr<StreamSocket>& socket,
+                                   Poco::MemoryInputStream& message)
+    {
+        assert(socket && "Must have a valid socket");
+
+        LOG_DBG("Check request: " << request.getURI());
+
+        bool validWopiConfig = false;
+        std::string connectMessage;
+        bool validConnectBack = false;
+
+        ConvertToPartHandler handler;
+        HTMLForm form(request, message, handler);
+
+        std::string url = (form.has("url") ? form.get("url") : "");
+        Poco::URI remoteServerURI(url);
+
+        // Validate allowed WOPI host
+        std::string addressToCheck = remoteServerURI.getHost();
+        LOG_DBG("Checking remote host: " << addressToCheck);
+        try
+            {
+                if (!StorageBase::allowedWopiHost(addressToCheck) && !net::isLocalhost(addressToCheck))
```

Comment:
We should not assume that localhost or 127.0.0.x are route-able and reachable locally - this is often not the case in containers - and the main problem we have with setups.